go/grpcwebproxy/main.go

```diff
@@ -48,8 +51,24 @@ func main() {
 	grpcServer := buildGrpcProxyServer(logEntry)
 	errChan := make(chan error)
 
-	// gRPC-Web compatibility layer with CORS configured to accept on every
-	wrappedGrpc := grpcweb.WrapServer(grpcServer, grpcweb.WithCorsForRegisteredEndpointsOnly(false))
+	if *flagCorsAllowAllOrigins && len(*flagCorsAllowOrigins) != 0 {
+		logrus.Fatal("Ambiguous cors_allow_all_origins and cors_allow_origins configuration. Either set cors_allow_all_origins=true OR specify one or more origins to whitelist with cors_allow_origins, not both.")
+	}
+
+	wrappedGrpc := grpcweb.WrapServer(grpcServer,
+		grpcweb.WithCorsForRegisteredEndpointsOnly(false),
+		grpcweb.WithOriginFunc(func(origin string) bool {
+			if *flagCorsAllowAllOrigins {
+				return true
+			}
+			for _, allowedOrigin := range *flagCorsAllowOrigins {
```

Comment:
Looping over all the flags will cause this function to run each time a request is received. Perhaps we can do a map lookup?

Reply:
Addressed in the latest round.